### PR TITLE
`VGSImageView`

### PR DESCRIPTION
**Pull request by DmytroDm**
## Description of changes
 - Add `VGSImageView`
 - Implement default image view attributes
 - Implement default image view functions
 - Handle image reveal
 - Add clear/hasImage functions
 - Add Javadoc
 - `ViewsStore` refactoring
 - Add unit tests

[CSDK-577]: https://verygoodsecurity.atlassian.net/browse/CSDK-577?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/verygoodsecurity/demoshow/ui/activity/ImageActivity.kt [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    private fun setupShow() {
+        show.subscribe(vgsImageView)
+        show.addOnResponseListener(this)
+
+        mbReveal.setOnClickListener {
+            flProgress.visibility = View.VISIBLE
+            show.requestAsync(
+                VGSRequest.Builder("post", VGSHttpMethod.POST)
+                    .body(mapOf("revealed_image" to (tvFileAlias.text ?: "tok_sandbox_9tXw3pQdLUq25Lm4bVpMkC")))
```
Comment (DonaldRG) [inline app/src/main/java/com/verygoodsecurity/demoshow/ui/activity/ImageActivity.kt:97]: I think this hardcoded token will never work `tok_sandbox_9tXw3pQdLUq25Lm4bVpMkC` maybe it is a better idea to show a message or something like that so the user knows it is failing?

File: app/src/main/java/com/verygoodsecurity/demoshow/ui/activity/PDFActivity.kt [hunk 2]
```diff
@@ -122,7 +122,7 @@ class PDFActivity : AppCompatActivity(), VgsCollectResponseListener, VGSOnRespon
             flProgress.visibility = View.VISIBLE
             show.requestAsync(
                 VGSRequest.Builder("post", VGSHttpMethod.POST)
-                    .body(mapOf("payment_card_pdf" to (tvFileAlias.text ?: "")))
+                    .body(mapOf("revealed_pdf" to "tok_sandbox_axTwt7EZLMHnTTFd8Ttaqe"))
```
Comment (DonaldRG) [inline app/src/main/java/com/verygoodsecurity/demoshow/ui/activity/PDFActivity.kt:125]: Same question here regarding the hardcoded token

File: vgsshow/src/main/java/com/verygoodsecurity/vgsshow/core/exception/VGSException.kt [hunk 3]
```diff
@@ -11,56 +11,65 @@ internal sealed class VGSException : Exception() {
...
+    internal class RequestTimeout : VGSException() {
 
         override val code: Int
             get() = 1483
 
         override val errorMessage: String = "TimeoutException"
     }
 
-    class ResponsePayloadException : VGSException() {
+    internal class ResponsePayload : VGSException() {
```
Comment (DonaldRG) [inline vgsshow/src/main/java/com/verygoodsecurity/vgsshow/core/exception/VGSException.kt:51]: Thanks for this refactor 👍

**Review by DonaldRG — COMMENTED** [review]
Everything looks really good, I only added a question regarding a hardcoded token.

File: app/src/main/java/com/verygoodsecurity/demoshow/ui/activity/ImageActivity.kt (hunk 1 above)
Comment (DmytroDm) [reply app/src/main/java/com/verygoodsecurity/demoshow/ui/activity/ImageActivity.kt:97]: @DonaldRG Those hardcoded token was added for testing, thanks.

**Review by dmytrokhl — APPROVED** [review]

**Review by DonaldRG — APPROVED** [review]
Everything looks good, thanks for the updates!

**Pull request merged**